packages/http-message-signatures/src/httpbis/index.ts

```diff
@@ -0,0 +1,338 @@
+            break
+          case 'keyid':
+            knownParameters.keyid = `${value.toString()}`
+            break
+          case 'alg':
+            knownParameters.alg = algorithmValue(value, key)
+            break
+        }
+      })
+
+      returnValue.set(signatureName, {
+        input: {
+          components,
+          parameters
+        },
+        value,
+        components: components.map(([component, params], i) => {
+          // @todo - params may be used to indicate that the componenet
+          // comes from the request when doing request/response binding
+          // See: https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-message-signatures-11#section-2.3
+          if (params.size > 0) {
+            throw new Error('Component parameters are not yet supported')
+          }
+          if (typeof component !== 'string') {
+            throw new Error(
+              `Error parsing component at index '${i}'. Expected a string but got a ${typeof component}.`
+            )
+          }
+          return component
+        }),
+        signatureParams,
+        ...knownParameters
+      })
+    }
+  )
+  return returnValue
+}
+
+export function buildSignedData(
+  request: RequestLike,
+  components: Component[],
+  signatureInputString: string
+): string {
+  const parts = components.map((component) => {
+    let value
+    if (component.startsWith('@')) {
+      value = extractComponent(request, component)
+    } else {
+      value = extractHeader(request, component)
+    }
+    return `"${component.toLowerCase()}": ${value}`
+  })
+  parts.push(`"@signature-params": ${signatureInputString}`)
+  return parts.join('\n')
+}
+
+// @todo - should be possible to sign responses too
+export async function sign(
+  request: RequestLike,
+  opts: SignOptions
+): Promise<RequestLike> {
+  const signingComponents: Component[] =
+    opts.components ?? defaultSigningComponents
+  const signingParams: Parameters = populateDefaultParameters({
+    ...opts.parameters,
+    keyid: opts.keyId,
+    alg: opts.signer.alg
+  })
+  const signatureInputString = buildSignatureInputString(
+    signingComponents,
+    signingParams
+  )
+  const dataToSign = buildSignedData(
+    request,
+    signingComponents,
+    signatureInputString
+  )
+  const signature = await opts.signer(Buffer.from(dataToSign))
+  Object.assign(request.headers, {
+    Signature: `sig1=:${signature.toString('base64')}:`,
+    'Signature-Input': `sig1=${signatureInputString}`
+  })
+  return request
+}
+
+export async function verify(
+  request: RequestLike,
+  opts: VerifyOptions
+): Promise<boolean> {
+  const signatures = parseSignatures(request)
+
+  signatures.forEach(
+    ({ value, components, signatureParams, keyid, alg }, signatureName) => {
+      if (!keyid) {
+        return false
+      }
+
+      const verifier = opts.verifiers[keyid]
+      if (!verifier || verifier.alg !== alg) {
+        return false
```

Comment:
`verify` doesn't appear to work as expected.
We're hitting this in our `'returns 200 with valid http signature'` test, but `verify` is returning true.

Reply:
Isn't it supposed to return true if the signature is valid?